python/lsst/validate/drp/validate.py

```diff
@@ -52,9 +55,113 @@ class bcolors(object):
     UNDERLINE = '\033[4m'
 
 
-def run(repo, dataIds, metrics, outputPrefix=None, level="design", verbose=False, **kwargs):
+def load_json_output(filepath):
+    """Read JSON from a file into a job object.
+
+    Currently just does a trivial de-serialization with no checking
+    to make sure that one results with a valid validate.base.job object.
+
+    Parameters
+    ----------
+    filepath : `str`
+        Source file name for JSON output.
+
+    Returns
+    -------
+    job : A `validate.base.job` object.
+    """
+    with open(filepath, 'r') as infile:
+        json_data = json.load(infile)
+
+    return Job.from_json(json_data)
+
+
+def get_filter_name_from_job(job):
+    """Get the filtername from a validate.base.job object
+
+    Assumes there is only one filter name and that it's the one in
+    the first measurement
+
+    Parameters
+    ----------
+    job : `validate.base.job` object
+
+    Returns
+    -------
+    filter_name : `str`
+    """
+    measurement_iterator = job.measurements
+    measurement = next(measurement_iterator)
+    filter_name = measurement.filter_name
+
+    return filter_name
+
+
+def run(repo_or_json, metrics=None, makePrint=True, makePlot=True,
+        level='design', **kwargs):
     """Main entrypoint from ``validateDrp.py``.
 
+    makePrint : bool, optional
+        Print calculated quantities (to stdout).
+    makePlot : bool, optional
+        Create plots for metrics.  Saved to current working directory.
+    level : str
+        Use <level> E.g., 'design', 'minimum', 'stretch'.
+
+    Arguments
+    ---------
+    repo : `str`
+        The repository.  This is generally the directory on disk
+        that contains the repository and mapper.
+
+        This can also be the filepath for a JSON file that contains
+        the cached output from a previous run.
+    """
+    if repo_or_json[-5:] == '.json':
```

Comment:
Typically you write this as ``if os.path.splitext(repo_or_json)[-1] == '.json'``